File: program/print-numbers-from-n-to-1-without-using-a-loop/PrintNumbersFromNTo1WithoutUsingALoop.cpp
// Write a program, that prints numbers from n to 1 without using a loop
#include <iostream>

// Instead of using a loop, this function just prints the current number and
// calls itself again with the number - 1(Recursion).
void PrintNum(int);

int main()
{
    std::cout << "Enter a num: ";
    int n;
    std::cin >> n;

    PrintNum(n);
    return 0;
}

void PrintNum(int n)
{
    // base case
    if (n > 0)
    {
        std::cout << n << " ";
        PrintNum(n - 1);
    }
    else
        std::cout << std::endl;

    return;
}

File: program/print-numbers-from-n-to-1-without-using-a-loop/PrintNumbersFromNTo1WithoutUsingALoop.cs
using System;

class MainClass {
    public static void Main (string[] args) {
     int i = 5;
     i = i+1;
     begin: 
     i = i - 1; 
     Console.Write(" " + i + " "); 
     if (i > 1) 
     { 
     goto begin; 
     } 
  }
}
